lib/event-highlighting/event-highlight-worker.js

```diff
@@ -0,0 +1,40 @@
+const EventMapHelper = require('./event-map-helper')
+
+require("../polyfills/set");
+
+let messageBuffer = new Map();
+let receivedThisFrame = new Map();
+
+self.onmessage = function(e) {
```

Comment:
I'm not really an expert in js workers, but I see that after the instantiation (`event-highlighter.js:52`, this `onmessage` field is overridden. Is then this assignment useless?

Reply:
Nope, this is the function that will be triggered, when the web worker receives messages from the main thread. The other one in in `event-highlighter.js:52` is the function that will be triggered, when the main thread receives a message from the web worker. 

This is the hole main idea. Doing the calculation and message buffer handling within a separate thread and independant from the animation frame.